dotcom-rendering/src/web/components/BlockquoteBlockComponent.tsx

```diff
@@ -34,52 +34,92 @@ const quotedBlockquoteStyles = (palette: Palette) => css`
 `;
 
-export const BlockquoteBlockComponent = ({ html, palette, quoted }: Props) => {
-	const {
-		willUnwrap: isUnwrapped,
-		unwrappedHtml,
-		unwrappedElement,
-	} = unwrapHtml({
-		fixes: [
-			{ prefix: '<p>', suffix: '</p>', unwrappedElement: 'p' },
-			{
-				prefix: '<blockquote>',
-				suffix: '</blockquote>',
-				unwrappedElement: 'blockquote',
-			},
-			{
-				prefix: '<blockquote class="quoted">',
-				suffix: '</blockquote>',
-				unwrappedElement: 'div',
-			},
-		],
-		html,
-	});
+const parseHtml = (html: string): DocumentFragment => JSDOM.fragment(html);
+
+// The nodeType for ELEMENT_NODE has the value 1.
+function isElement(node: Node): node is Element {
+	return node.nodeType === 1;
+}
 
-	if (quoted) {
-		const htmlWithIcon = unwrappedHtml
-			.trim()
-			.replace(
-				'<p>',
-				`<p>${renderToString(
-					<QuoteIcon colour={palette.fill.blockquoteIcon} />,
-				)}`,
-			);
-		return (
-			<RewrappedComponent
-				isUnwrapped={isUnwrapped}
-				html={htmlWithIcon}
-				tagName="blockquote"
-				elCss={quotedBlockquoteStyles(palette)}
-			/>
+const getAttrs = (node: Node): NamedNodeMap | undefined =>
+	isElement(node) ? node.attributes : undefined;
+
+const textElement =
+	(isQuoted: boolean, palette: Palette) =>
+	(node: Node, key: number): ReactNode => {
+		const text = node.textContent ?? '';
+		const children = Array.from(node.childNodes).map(
+			textElement(isQuoted, palette),
 		);
-	}
-	return (
-		<RewrappedComponent
-			isUnwrapped={isUnwrapped}
-			html={unwrappedHtml}
-			elCss={simpleBlockquoteStyles}
-			tagName={unwrappedElement}
-		/>
-	);
+		switch (node.nodeName) {
+			case 'P': {
+				// We want to add the quote icon to the first child (p) of the blockquote element
+				if (isQuoted && node.parentElement?.nodeName === 'BLOCKQUOTE') {
+					return (
+						<p>
+							<QuoteIcon colour={palette.fill.blockquoteIcon} />
+							{children}
+						</p>
+					);
+				}
+				return h('p', { children });
+			}
+			case '#text':
+				return text;
+			case 'SPAN':
+				return text;
+			case 'A':
+				return h('A', {
+					href: getAttrs(node)?.getNamedItem('href'),
+					key,
+					children,
+				});
+			case 'BLOCKQUOTE':
+				return h('blockquote', {
+					key,
+					children,
+					css: isQuoted
+						? quotedBlockquoteStyles(palette)
+						: simpleBlockquoteStyles,
+				});
+			case 'STRONG':
+				return h('strong', {
+					css: { fontWeight: 'bold' },
+					key,
+					children,
+				});
+			case 'B':
+				return h('b', { key, children });
+			case 'EM':
+				return h('em', { key, children });
+			case 'BR':
+				return h('br', { key, children });
+			case 'UL':
+				return h('ul', { key, children });
+			case 'OL':
+				return h('ol', { key, children });
+			case 'LI':
+				return h('li', { key, children });
+			case 'MARK':
+				return h('mark', { key, children });
+			case 'SUB':
+				return h('sub', { key, children });
+			case 'SUP':
+				return h('sup', {
+					key,
+					children,
+				});
```

Comment:
This may make it harder to parse the differences between these cases, so you may want to do something along the lines of:

```typescript
            // …
			case 'B':
			case 'EM':
			case 'BR':
			case 'UL':
			case 'OL':
			case 'LI':
			case 'MARK':
			case 'SUB':
			case 'SUP':
			    // you don’t even need `.toLowerCase()`
				return h(node.nodeName, {
					key,
					children,
				});
			// …
``` 

[Working example with uppercase](https://preactjs.com/repl?code=aW1wb3J0IHugcmVuZGVyIH0gZnJvbSAncHJlYWN0JzsKCmNvbnN0IEV4YW1wbGUgPSAoKSA9PiBoKCJQIiwge30sCgkJCQkJICJUaGlzIGlzIGEgcGFyYWdyYXBoLiAiLAoJCQkJCSBoKCJFTSIsIHt9LAoJCQkJCQkgIlRoaXMgaXMgaW4gaXRhbGljcyAiLAoJCQkJCQkgaCgiU1RST05HIiwge30sICJhbmQgYm9sZCIpCgkJCQkJCSkKCQkJCQkpOwoKcmVuZGVyKDxFeGFtcGxlIC8%2BLCBkb2N1bWVudC5nZXRFbGVtZW50QnlJZCgnYXBwJykpOwo%3D)

Reply:
Good idea :)